### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by giraffate**
changelog: none

### CODE REVIEW

**Comment by giraffate** [comment]
Hmm, test `clippy-driver` seems fail. Sorry, I can't investigate immediately. If anyone can handle this soon, please close this PR.

File: .github/driver.sh [hunk 1]
```diff
@@ -22,9 +22,9 @@ unset CARGO_MANIFEST_DIR
 
 # Run a lint and make sure it produces the expected output. It's also expected to exit with code 1
 # FIXME: How to match the clippy invocation in compile-test.rs?
-./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cstring.rs 2> cstring.stderr && exit 1
-sed -e "s,tests/ui,\$DIR," -e "/= help/d" cstring.stderr > normalized.stderr
-diff normalized.stderr tests/ui/cstring.stderr
+./target/debug/clippy-driver -Dwarnings -Aunused -Zui-testing --emit metadata --crate-type bin tests/ui/cast.rs 2> cast.stderr && exit 1
+sed -e "s,tests/ui,\$DIR," -e "/= help/d" cast.stderr > normalized.stderr
+diff normalized.stderr tests/ui/cast.stderr
```
Comment (matthiaskrgr) [inline .github/driver.sh:27]: can you change the `diff` call to `diff -u ...`?
This should make things more human readable (it'll be in the same format `git diff` that way)

Comment (giraffate) [reply .github/driver.sh:27]: For example, if I use `double_neg.stderr` not `cast.stderr`, the test seems to succeed. I still don't know if the test here should be `cast.stderr`.

Comment (giraffate) [reply .github/driver.sh]: Looking at https://github.com/rust-lang/rust-clippy/pull/6257/commits/a1bb10e9b805d3bd60d7c47dc453c011478245df, I didn't know if it had to be `cast.stderr` in the test here, so I use `double_neg.stderr`. If it's not just testing in one file and  it has to be `cast.stderr`,  I would consider modifying the sed regex.

**Comment by ebroto** [comment]
@bors r+

Thanks for doing this!

I've amended the PR to deprecate the lint instead of just removing it. Otherwise users that were referring to the lint (e.g. allowing it) would get an error saying the lint did not exist. By deprecating it we change that to a more informative warning.

I will sync that change back to rustc.

**Comment by giraffate** [comment]
> I've amended the PR to deprecate the lint instead of just removing it. Otherwise users that were referring to the lint (e.g. allowing it) would get an error saying the lint did not exist. By deprecating it we change that to a more informative warning.

Thanks!

**Pull request merged**